src/main/java/com/google/devtools/build/lib/rules/genrule/GenRuleBase.java

```diff
@@ -321,6 +326,31 @@ public String lookupVariable(String variableName) throws ExpansionException {
         return expandSingletonArtifact(filesToBuild, "$@", "output file");
       }
 
+      if (variableName.equals("OUT_ROOT_DIR")) {
+        // The output root directory. This variable expands to the package's root directory
+        // in the genfiles tree.
+        Path dirPath;
+        PathFragment dirFragment;
+        if (ruleContext.getRule().hasBinaryOutput()) {
+          dirPath = ruleContext.getConfiguration().getBinDir().getRoot().asPath();
+          dirFragment = ruleContext.getConfiguration().getBinFragment();
+        } else {
+          dirPath = ruleContext.getConfiguration().getGenfilesDir().getRoot().asPath();
+          dirFragment = ruleContext.getConfiguration().getGenfilesFragment();
+        }
+
+        PathFragment relPath = ruleContext.getRule().getLabel().getPackageIdentifier().getSourceRoot();
+        // TODO: Instead of creating a new directory, use the root
+        String outDirStr = "testing-genrules";
```

Comment:
What's this for?

Doing I/O is not a good idea during the analysis phase, because the results of analysis are sometimes cached and I/O side effects are not accounted for. Thus, e.g. if this directory was deleted from outside of Bazel, the rule would not get re-run and the output tree would be an inconsistent state.

(that's why throwing `IOException` is not allowed here)

Reply:
Significantly simplified this code to just point to the root directory instead of creating a new one.